UnityProject/Assets/Scripts/Player/MovementV2/MatrixCash.cs

```diff
@@ -95,8 +95,10 @@ public void ResetNewPosition(Vector3 Centrepoint, RegisterTile Inon)
 			Positions[i] = null;
 		}
 
-		Positions[6] = Inon.Matrix.MatrixInfo;
-
+		if (Positions != null && Inon.OrNull()?.Matrix.OrNull()?.MatrixInfo is not null)
```

Comment:
tthis should be catch since it's to do with movement?

Reply:
Updated to include a log